core/src/main/java/com/graphhopper/routing/util/spatialrules/SpatialRuleLookupBuilder.java

```diff
@@ -53,27 +54,27 @@ public static SpatialRuleLookup buildIndex(List<JsonFeatureCollection> jsonFeatu
                 List<Polygon> borders = new ArrayList<>();
                 for (int i = 0; i < jsonFeature.getGeometry().getNumGeometries(); i++) {
                     Geometry poly = jsonFeature.getGeometry().getGeometryN(i);
-                    if (poly instanceof Polygon)
-                        borders.add((Polygon) poly);
-                    else
+                    if (poly instanceof Polygon) {
+                        Geometry intersection = bboxGeometry.intersection(poly);
+                        if (!intersection.isEmpty()) {
+                            PolygonExtracter.getPolygons(intersection, borders);
```

Comment:
Why is this necessary and not just `borders.add(intersection)`? Isn't intersection always a Polygon?

Reply:
[Geometry.intersection(Geometry other)](https://locationtech.github.io/jts/javadoc/org/locationtech/jts/geom/Geometry.html#intersection-org.locationtech.jts.geom.Geometry-):
> If the result is empty, it is an atomic geometry with the dimension of the lowest input dimension

For two polygons this would result in an empty polygon being picked up by the PolygonExtracter.